### PR TITLE
Minor changes

### PR DESCRIPTION
**Pull request by vitartas**
- Added pool.close() and pool.join() when using Pool to reduce the number of ASE.jl errors
- Fixed two small bugs in AL
- Changed the default metadynamics 2d plot format to .pdf
- Added type hints to private functions

### CODE REVIEW

**Comment by t-young31** [comment]
would you mind PR-ing the first 5 commits, then the black one to make it easier to review?

**Review by tristan-j-wood — APPROVED** [review]

**Review by tristan-j-wood — APPROVED** [review]

File: mlptrain/training/active.py [hunk 1]
```diff
@@ -550,10 +566,15 @@ def _attach_plumed_coords_to_init_configs(init_configs, bias) -> None:
     return None
 
 
-def _update_init_config(init_config, mlp, fix_init_config, bias,
-                        inherit_metad_bias, bias_start_iter, iteration
+def _update_init_config(init_config: 'mlptrain.Configuration',
+                        mlp: 'mlptrain.potentials._base.MLPotential',
+                        fix_init_config: bool,
+                        bias: Optional,
```
Comment (t-young31) [inline mlptrain/training/active.py:572]: missing type

File: mlptrain/configurations/configuration_set.py [hunk 2]
```diff
@@ -617,8 +617,10 @@ def _run_parallel_method(self, function, **kwargs):
                                           kwds=kwargs)
                 results.append(result)
 
+            pool.close()
             for i, result in enumerate(results):
                 self[i] = result.get(timeout=None)
+            pool.join()
```
Comment (t-young31) [inline mlptrain/configurations/configuration_set.py:623]: I didn't think you needed to do this with a mp pool 😕 . doc: https://docs.python.org/3/library/multiprocessing.html#using-a-pool-of-workers

have you got a test that fails without this and succeeds with this?

**Comment by t-young31** [comment]
thanks for all the functions call/signature fixes ❤️

File: mlptrain/configurations/configuration_set.py (hunk 2 above)
Comment (vitartas) [reply mlptrain/configurations/configuration_set.py:623]: We are sometimes getting errors when using linear ACE and we are not entirely sure why. Most of them arise when performing tasks in parallel, and we've found that adding pool.close() and pool.join() relieves some of these errors.

**Pull request merged**